third_party/2and3/werkzeug/test.pyi

```diff
@@ -67,28 +70,63 @@ class EnvironBuilder:
 
 class ClientRedirectError(Exception): ...
 
-class Client:
+_ResponseWrapper = TypeVar('_ResponseWrapper', bound=Response)
+
+class Client(Generic[_ResponseWrapper]):
     application: Any
-    response_wrapper: Any
+    response_wrapper: Type[_ResponseWrapper]
     cookie_jar: Any
     allow_subdomain_redirects: Any
-    def __init__(self, application, response_wrapper: Optional[Any] = ..., use_cookies: bool = ...,
+    def __init__(self, application, response_wrapper: Optional[Type[_ResponseWrapper]] = ..., use_cookies: bool = ...,
                  allow_subdomain_redirects: bool = ...): ...
     def set_cookie(self, server_name, key, value: str = ..., max_age: Optional[Any] = ..., expires: Optional[Any] = ...,
                    path: str = ..., domain: Optional[Any] = ..., secure: Optional[Any] = ..., httponly: bool = ...,
                    charset: Text = ...): ...
     def delete_cookie(self, server_name, key, path: str = ..., domain: Optional[Any] = ...): ...
     def run_wsgi_app(self, environ, buffered: bool = ...): ...
     def resolve_redirect(self, response, new_location, environ, buffered: bool = ...): ...
-    def open(self, *args, **kwargs): ...
-    def get(self, *args, **kw): ...
-    def patch(self, *args, **kw): ...
-    def post(self, *args, **kw): ...
-    def head(self, *args, **kw): ...
-    def put(self, *args, **kw): ...
-    def delete(self, *args, **kw): ...
-    def options(self, *args, **kw): ...
-    def trace(self, *args, **kw): ...
+
+    @overload
+    def open(self, *args, as_tuple: Literal[True], **kwargs) -> Tuple[Mapping[str, Any], _ResponseWrapper]: ...
+    @overload
+    def open(self, *args, as_tuple: Literal[False] = ..., **kwargs) -> _ResponseWrapper: ...
```

Comment:
I am not very experienced with using `Literal` myself, but from my short tests, we need another overload for `bool` in general for cases like this:

```python
def foo(x: bool):
    client.open(as_tuple=x)
```

Should just return `Any`.

(Same for the methods below.)

Reply:
Yep, you're right. Done.